cpp/vcpkg.json

```diff
@@ -49,6 +54,7 @@
     "rapidjson",
     "re2",
     "snappy",
+    "spdlog",
```

Comment:
Note to myself: After this PR is merged, I will make comment on https://github.com/apache/arrow/pull/40939:

please review the `cpp/vcpkg.json` changes